src/actions/bump.ts

```diff
@@ -145,103 +123,45 @@ async function run(): Promise<void> {
-      core.endGroup();
-      if (release || tag) {
-        const relType = tag ? "tag" : "release";
-        if (!isBranchAllowedToPublish) {
-          core.startGroup(`ℹ️ Branch ${branchName} is not allowed to publish`);
-          core.info(
-            `Only branches that match the following ECMA-262 regular expression may publish:\n${allowedBranchesRegEx}`
-          );
-        } else if (isPullRequestEvent()) {
-          core.startGroup(
-            `ℹ️ Not creating ${relType} on a pull request event.`
-          );
-          core.info(
-            "We cannot create a release or tag in a pull request context, due to " +
-              "potential parallelism (i.e. races) in pull request builds."
-          );
-        } else {
-          core.startGroup(`ℹ️ Creating ${relType} ${nv}..`);
-          try {
-            if (tag) {
-              await createTag(nv, context.sha);
-            } else {
-              const changelog = await generateChangelog(bumpInfo);
-              await createRelease(nv, context.sha, changelog);
-            }
-          } catch (ex: unknown) {
-            // The most likely failure is a preexisting tag, in which case
-            // a RequestError with statuscode 422 will be thrown
-            const commit = await getShaForTag(`refs/tags/${nv}`);
-            if (ex instanceof RequestError && ex.status === 422 && commit) {
-              core.setFailed(
-                `Unable to create ${relType}; the tag "${nv}" already exists in the repository, ` +
-                  `it currently points to ${commit}.\n` +
-                  "You can find the branch(es) associated with the tag with:\n" +
-                  `  git fetch -t; git branch --contains ${nv}`
-              );
-            } else if (ex instanceof RequestError) {
-              core.setFailed(
-                `Unable to create ${relType} with the name "${nv}" due to ` +
-                  `HTTP request error (status ${ex.status}):\n${ex.message}`
-              );
-            } else if (ex instanceof Error) {
-              core.setFailed(
-                `Unable to create ${relType} with the name "${nv}":\n${ex.message}`
-              );
-            } else {
-              core.setFailed(`Unknown error during ${relType} creation`);
-              throw ex;
-            }
-            core.endGroup();
-            return;
-          }
-          core.info("Succeeded");
-        }
-      } else {
-        core.startGroup(`ℹ️ Not creating tag or release for ${nv}..`);
-        core.info(
-          "To create a lightweight Git tag or GitHub release when the version is bumped, run this action with:\n" +
-            ' - "create-release" set to "true" to create a GitHub release, or\n' +
-            ' - "create-tag" set to "true" for a lightweight Git tag.\n' +
-            "Note that setting both options is not needed, since a GitHub release implicitly creates a Git tag."
+    if (config.versionScheme === "semver") {
```

Comment:
Preference; use a `switch`

Reply:
Not my preference, so @maikelvdh's vote has it ;)